providers/runtime.go

```diff
@@ -232,7 +235,9 @@ func (r *Runtime) Connect(req *plugin.ConnectReq) error {
 
 	// }
 
+	r.mu.Lock()
 	r.Provider.Connection, r.Provider.ConnectionError = r.Provider.Instance.Plugin.Connect(req, &callbacks)
+	r.mu.Unlock()
```

Comment:
The problem is the assignation here, not the actual gRPC call `Connect()`. I will change this.